plan/logical_plan_builder.go

```diff
@@ -1318,6 +1322,27 @@ func (b *planBuilder) projectVirtualColumns(ds *DataSource, columns []*table.Col
 	return proj
 }
 
+// wrapCastOnExpr wraps a CAST operation on the expression. It's used for generated columns
+// because the generation expression may has different field type with the column respectively.
+func wrapCastOnExpr(expr expression.Expression, ft *types.FieldType, ctx context.Context) expression.Expression {
```

Comment:
can we use `expression.buildCastFunction` here?

Reply:
That function is private in expression. So I will try make it public.